server/src/instant/db/indexing_jobs.clj

```diff
@@ -245,16 +268,19 @@
   (let [attrs (attr-model/get-by-app-id conn app_id)
         etype (attr-model/fwd-etype (attr-model/seek-by-id attr_id attrs))
         _ (assert etype "Attribute has no etype")
-        id-attr-id (:id (attr-model/seek-by-fwd-ident-name [etype "id"] attrs))
-        _ (assert id-attr-id (str etype " has no id attribute"))
+        id-attr (attr-model/seek-by-fwd-ident-name [etype "id"] attrs)
+        _ (assert id-attr (str etype " has no id attribute"))
         indexed-attr (attr-model/seek-by-id attr_id attrs)
         _ (assert indexed-attr (str "no attr found with id " attr_id))]
     (if (not= (:value-type indexed-attr) :blob)
       ;; Just return false if it's not a blob
       [:= [:inline 1] [:inline 0]]
       [:and
        [:= :triples.app-id app_id]
-       [:= :triples.attr-id id-attr-id]
+       [:= :triples.attr-id (:id id-attr)]
+       (when (and (:unique? id-attr)
```

Comment:
Applies the same optimization from the required check: https://github.com/instantdb/instant/pull/1221/commits/8e0e60f2e1362f9614f47628dccabb1ca74dcb62#diff-cdcec9fc9613966c0a6a5b3a6047e47ff655a36b1613164fcf49a78f385fb068R810